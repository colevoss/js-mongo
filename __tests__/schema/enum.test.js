const Schema = require('../../src/Schema');

describe('Schema.enum', () => {
  test('Returns valid option of the given enum', () => {
    const testData = { test: 'TEST' };

    expect(Schema.enum('TEST', 'OTHER_TEST')('test', testData)).toBe(
      testData.test,
    );
  });

  test('Returns error', () => {
    const testData = { test: 'NOT_TEST' };

    const checked = Schema.enum('TEST', 'OTHER_TEST')('test', testData);

    expect(checked).toBeInstanceOf(TypeError);
    expect(checked).toMatchSnapshot();
  });

  test('__type', () => {
    expect(Schema.enum('test', 'test 2').__type).toBe('enum');
  });
});
