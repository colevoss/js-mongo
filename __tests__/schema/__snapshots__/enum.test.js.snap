// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.enum Returns error 1`] = `[TypeError: Expected one of ["TEST","OTHER_TEST"] for test but received NOT_TEST]`;
